test/refactor_nrepl/ns/suggest_libspecs_test.clj

```diff
@@ -0,0 +1,110 @@
+    ;; `preferred-aliases` are taken into account when they don't conflict with the `project-libspecs`:
+    "set"        "cljc"      "cljc"     [["set" "clojure.set"]]              '{:clj  {x [y]}
+                                                                               :cljs {x [y]}}                ["[clojure.set :as set]"]
+
+    ;; Story 3:
+    "set"        "cljc"      "cljc"     [["set" "clojure.set"]]              '{:clj  {set [something-else
+                                                                                           clojure.set]}
+                                                                               :cljs {set [something-else
+                                                                                           clojure.set]}}    ["[something-else :as set]" "[clojure.set :as set]"]
+
+    ;; Story 4:
+    "io"         "clj"       "clj"     [["io" "clojure.java.io" :only :clj]] '{:cljs {io [something-else]}}  ["[clojure.java.io :as io]"]
+    ;; Story 5:
+    "io"         "clj"       "clj"     [["io" "clojure.java.io" :only :clj]] '{:clj  {io [something-else]}}  ["[something-else :as io]"]
+    "io"         "clj"       "clj"     [["io" "something-else" :only :cljs]] {}                              ["[clojure.java.io :as io]"]
+    #_lib-prefix #_buffer-lc #_input-lc #_preferred-aliases                  #_project-libspecs              #_expected
+    ;; Story 6:
+    "io"         "cljc"      "cljc"    [["io" "clojure.java.io" :only :clj]] '{:cljs {io [test-cljc-ns]}}     ["[test-cljc-ns :as io]"
+                                                                                                               "#?(:clj [clojure.java.io :as io]\n      :cljs [test-cljc-ns :as io])"]
+    "io"         "cljc"      "clj"     [["io" "clojure.java.io" :only :clj]] '{:cljs {io [test-cljc-ns]}}     ["[test-cljc-ns :as io]"
+                                                                                                               "#?(:clj [clojure.java.io :as io]\n      :cljs [test-cljc-ns :as io])"]
+    "io"         "cljc"      "cljs"    [["io" "clojure.java.io" :only :clj]] '{:cljs {io [test-cljc-ns]}}     ["[test-cljc-ns :as io]"]
+    "io"         "cljc"      "clj"     [["io" "clojure.java.io" :only :clj]] '{:clj  {io [test-clj-ns]}}      ["#?(:clj [test-clj-ns :as io])"]
+    ;; The difference here is that test-cljs-ns is backed by a .cljs extension and therefore is not a valid .cljc or :clj suggestion
+    "io"         "cljc"      "cljc"    [["io" "clojure.java.io" :only :clj]] '{:cljs {io [test-cljs-ns]}}     ["#?(:clj [clojure.java.io :as io]\n      :cljs [test-cljs-ns :as io])"]
+    ;; Returns an empty form for the :clj branch when there's no valid suggestion for it:
+    "io"         "cljc"      "cljc"    []                                    '{:cljs {io [test-cljs-ns]}}     ["#?(:clj [ :as io]\n      :cljs [test-cljs-ns :as io])"]
```

Comment:
I'd rather default to not introducing errors. This is particularly an issue for figwheel/shadow-js setups in ClojureScript, as often the entire file is re-evaluated at save (instead of eval s-expr), which makes introducing errors pretty harsh. I know for me I would find the addition of the context specific require helpful, but would be very frustrated to need to clean up the empty clj case everytime I used it. 

I'm not against having this error as an option, but we are getting kinda complicated to test here so would prefer we keep it simplified for now, and maybe add the option to error later once the rest of this is settled?